index.js

```diff
@@ -79,70 +79,72 @@ Collector.prototype.computePointCountToSend = function(pointSizes, upperBound) {
   return Math.min(index, pointSizes.length); // But not if there were no points at all.
 };
 
-function notifyIfFlushed(collector, callback, err) {
-    if (!callback || collector._flushesInFlight > 0) {
+Collector.prototype._notifyIfFlushed = function(callback, err) {
+    var self = this;
+    if (!callback || self._flushesInFlight > 0) {
         return;
     }
     setImmediate(callback, err);
-}
+};
 
-function flushSeries(collector, seriesName, points, callback) {
+Collector.prototype._flushSeries = function(seriesName, points, callback) {
     if (!points || points.length === 0) {
         return;
     }
+    var self = this;
 
     // only send N points at a time to avoid making requests too large
     var spliceIndex;
-    if (collector._client.protocol == 'udp:') {
-      spliceIndex = collector.computePointCountToSend(collector._points.map(function (point) {
+    if (self._client.protocol == 'udp:') {
```

Comment:
A bunch of the `self`s here can be `this` instead - I don't think we have a policy on that so I guess either way works.